packages/login-with-qq/index.ts

```diff
@@ -0,0 +1,70 @@
+//Power by air. <air@adteam.cc>
+import {
+    Context, ForbiddenError, Handler, superagent, SystemModel,
+    TokenModel, UserFacingError, ValidationError, OauthModel, OauthCallbackHandler, 
+} from 'hydrooj';
+
+import { Notification } from '@hydrooj/ui-default';
+
+const { PRIV } = global.Hydro.model.builtin; // 内置 Privilege 权限节点
+
+declare module 'hydrooj' {
+    interface SystemKeys {
+        'login-qq.id': string,
+        'login-qq.secret': string,
+    }
+}
+
+async function get(this: Handler) {
+    const [appid, url, [state]] = await Promise.all([
+        SystemModel.get('login-qq.id'),
+        SystemModel.get('server.url'),
+        TokenModel.add(TokenModel.TYPE_OAUTH, 600, { redirect: this.request.referer }),
+    ]);
+    // eslint-disable-next-line max-len
+    this.response.redirect = `https://graph.qq.com/oauth2.0/authorize?response_type=code&client_id=${appid}&redirect_uri=${url}oauth/qq/callback&state=${state}`;
+}
+
+function unescapedString(escapedString: string) {
+    escapedString += new Array(5 - (escapedString.length % 4)).join('=');
+    return escapedString.replace(/-/g, '+').replace(/_/g, '/');
+}
+
+async function callback(this: Handler, { state, code} ) {
+    const [[appid, secret, url], s] = await Promise.all([
+        SystemModel.getMany([
+            'login-qq.id', 'login-qq.secret', 'server.url',
+        ]),
+        TokenModel.get(state, TokenModel.TYPE_OAUTH),
+    ]);
+    const res = await superagent.get(`https://graph.qq.com/oauth2.0/token?grant_type=authorization_code&client_id=${appid}&client_secret=${secret}&code=${code}&redirect_uri=${url}oauth/qq/callback&fmt=json`)
+        .set('accept', 'application/json')
+        .send();
```

Comment:
Please read the document of superagent.

Reply:
About QQ connect, this API does NOT support the use of the send method, only supports constructing url. 
Use send as follows：
`const res = await superagent.get(`https://graph.qq.com/oauth2.0/token`)
        .set('User-Agent', 'ADTeam-OAuth')
        .set('accept', 'application/json')
        .send({
            grant_type: `authorization_code`,
            client_id: appid,
            client_secret: secret,
            code,
            redirect_uri: `${url}oauth/qq/callback`,
            fmt: `json`,
        });`

When logging in, the API returns:
`callback( {"error":100004,"error_description":"param grant_type is wrong or lost "} );`

Reply:
Read the documention **CAREFULLY**.